pylint/checkers/utils.py

```diff
@@ -1172,8 +1172,12 @@ def class_is_abstract(node: nodes.ClassDef) -> bool:
         if meta.name == "ABCMeta" and meta.root().name in ABC_MODULES:
             return True
 
-    for ancestor in node.ancestors():
-        if ancestor.name == "ABC" and ancestor.root().name in ABC_MODULES:
+    # As well as directly inherited ABC class
+    for base in node.bases:
+        base_class_name = (
+            base.attrname if isinstance(base, astroid.Attribute) else base.as_string()
+        )
```

Comment:
I could not find any way to detect the module directly if you subclass ABC like this:

```
from abc import ABC
class MyTestCase(ABC):
  pass
````

Let me know if anyone has any ideas

Reply:
This is what we use `inference` for. You can look for calls to `safe_infer` which should hopefully set you on the right track!